webapi/src/main/scala/org/knora/webapi/messages/admin/responder/usersmessages/UsersMessagesADM.scala

```diff
@@ -837,13 +837,15 @@ case class UserChangeRequestADM(
     throw BadRequestException("Too many parameters sent for system admin membership change.")
   }
 
-  // change project memberships
-  if (projects.isDefined && parametersCount > 1) {
+  // change project memberships (could also involve changing projectAdmin memberships)
+  if (
+    projects.isDefined && projectsAdmin.isDefined && parametersCount > 2 || projects.isDefined && !projectsAdmin.isDefined && parametersCount > 1
```

Comment:
personally, I find such boolean expressions horrible to read/understand.  
I think some parentheses (and maybe a strategic line break) would already make it more readable.

another pattern I have come to like is using pattern matching, something along the lines of:
```scala
(project.isDefined, projectAdmin.isDefined, parametersCount) match {
  case (true, true, count) if count > 2 => ...
  case (true, false, count) if count > 1 => ...
}
```

Reply:
Me too ;) I just followed the pattern that was already there. IMO, we shouldn't validate the message by counting the parameters at all but use proper validation instead (with value objects). I thought this is out of scope here.